client/state/concierge/appointment-details/reducer.js

```diff
@@ -0,0 +1,17 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import { createReducer } from 'state/utils';
+import {
+	CONCIERGE_APPOINTMENT_DETAILS_REQUEST,
+	CONCIERGE_APPOINTMENT_DETAILS_UPDATE,
+} from 'state/action-types';
+
+export const appointmentDetails = createReducer( null, {
+	[ CONCIERGE_APPOINTMENT_DETAILS_REQUEST ]: () => null,
+	[ CONCIERGE_APPOINTMENT_DETAILS_UPDATE ]: ( state, action ) => action.appointmentDetails,
```

Comment:
Although it works great in this case, I feel it is not clear that our reducer loses the concept of appointment id here: Our query component for fetching and the action creators all contain an "id" field, which indicate that we are accessing a certain appointment data. Thus, I think it would be a bit surprising to the future maintainer that the reducer can only hold one appointment data at a time.

Could we consider to use `keyedReducer()` here and add an `appointmentId` parameter to the selector?

Reply:
Thanks for taking the time, great suggestion, updated.